### PR TITLE
code challenge 12 complete

### PR DESCRIPTION
**Pull request by carlydekock**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
   - [x] “Happy Path” - Expected outcome
   - [x] Expected failure
   - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
   - [x] Summary, Description, Approach & Efficiency, Solution
   - [x] Link to code

### CODE REVIEW

File: challenges/fifoAnimalShelter/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - dequeue(pref): returns either a dog or a cat. If pref is not "dog" or "cat" then return null
+
+## Approach & Efficiency
+<!-- What approach did you take? Why? What is the Big O space/time for this approach? -->
+- Enqueue: space O(1)/time O(1)
+- Dequeue: space O(1)/time O(1)
+
+## Resources and Collaborators
+
+- Worked at lab table with Jason Q, Jason D, Nick M
```
Comment (adrianhuebner) [inline challenges/fifoAnimalShelter/README.md:19]: Would love to see whiteboard for challenges like these
